application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java

```diff
@@ -0,0 +1,103 @@
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.extension.ExtensionId;
+import org.xwiki.extension.InstalledExtension;
+import org.xwiki.extension.UninstallException;
+import org.xwiki.extension.event.ExtensionUninstalledEvent;
+import org.xwiki.extension.repository.InstalledExtensionRepository;
+import org.xwiki.model.namespace.WikiNamespace;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Listens to extension uninstall events and if the Admin Tools extension is being uninstalled, attempt to remove the
+ * remaining modules.
+ *
+ * @version $Id$
+ */
+@Component
+@Named(AdminToolsUninstallListener.HINT)
+@Singleton
+public class AdminToolsUninstallListener extends AbstractEventListener
+{
+    /**
+     * The hint for the component.
+     */
+    public static final String HINT = "AdminToolsUninstallListener";
+
+    @Inject
+    protected Logger logger;
+
+    @Inject
+    private Provider<XWikiContext> wikiContextProvider;
+
+    @Inject
+    private InstalledExtensionRepository installedRepository;
+
+    /**
+     * Creates an event-listener filtering for ExtensionUninstalledEvent.
+     */
+    public AdminToolsUninstallListener()
+    {
+        super(HINT, new ExtensionUninstalledEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        if (event instanceof ExtensionUninstalledEvent) {
+            try {
+                ExtensionId uninstalledExtension = ((ExtensionUninstalledEvent) event).getExtensionId();
+                if (uninstalledExtension.getId().equals(getExtensionId("ui"))) {
+                    ExtensionId apiExtensionId =
+                        new ExtensionId(getExtensionId("api"), uninstalledExtension.getVersion());
+                    ExtensionId defaultExtensionId =
+                        new ExtensionId(getExtensionId("default"), uninstalledExtension.getVersion());
+
+                    InstalledExtension apiExtension = installedRepository.getInstalledExtension(apiExtensionId);
+                    InstalledExtension defaultExtension = installedRepository.getInstalledExtension(defaultExtensionId);
+                    String namespace = new WikiNamespace(wikiContextProvider.get().getWikiId()).serialize();
+                    installedRepository.uninstallExtension(defaultExtension, namespace);
+                    installedRepository.uninstallExtension(apiExtension, namespace);
+                    logger.info("Successfully uninstalled all Admin Tools modules.");
+                }
+            } catch (UninstallException e) {
+                logger.error("There was an error while uninstalling Admin Tools modules.");
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private String getExtensionId(String module)
```

Comment:
`getAdminToolsExtensionId`. I know it's longer, but it was too generic
Maybe `getATExtensionId`? I know it's a common practice to use acronyms for extensions when the name it's too long

Reply:
I think `getAdminToolsExtensionId` it's fine. It does not break any line and it's clear.